gdp-common-utilities/src/main/java/gov/usgs/cida/gdp/utilities/TimeRangeUtil.java

```
@@ -0,0 +1,102 @@
+                    range = rangeList.get(0);
+                } else {
+                    // decomposing along a single range for axes with more than
+                    // one is something we don't want to do.  This should only
+                    // happen with X and Y axes (and this method shouldn't be
+                    // called for those)
+                    throw new IllegalArgumentException(
+                            "Axis \"" + axis.getFullName() + "\" has coordinates with more that 1 dimension");
+                }
+            }
+        }
+        return decomposeRange(range);
+    }
+    
+    public static Iterable<Range> decomposeRange(Range range) {
+        return new SingleElementRangeDecomposer(range);
+    }
+    
+    public static class SingleElementRangeDecomposer implements Iterable<Range> {
+        public final Range range;
+        public SingleElementRangeDecomposer(Range range) {
+            this.range = range;
+        }
+        @Override
+        public Iterator<Range>iterator() {
+            return range == null ?
+                Arrays.asList((Range)null).iterator() :
+                new SingleElementRangeIterator(range);
+        }
+        public class SingleElementRangeIterator implements Iterator<Range> {
+            private final Range range;
+            private int next = 0;
+            public SingleElementRangeIterator(Range range) {
+                this.range = range;
+            }
+            @Override
+            public boolean hasNext() {
+                return next < range.length();
+            }
+            @Override
+            public Range next() {
+                Range r = null;
+                try {
+                    int e = range.element(next++);
+                    // yes!  upper-bound is inclusive!  Yes I agree it's wierd!
+                    r = new Range(e, e, 1);
+                } catch (InvalidRangeException ignore) { /* why is this a checked exception? */ }
```

Is there any problem ignoring this exception? Will we ever hit this and end up with a null Range?